CHANGELOG

```diff
@@ -12,6 +12,12 @@
 Version 4.2.1 (development)
 ===========================
 
+- Changed the auto refinement algorithm, which is now based on the order of
+  the grid function and mesh. The total number of elements is limited to the
+  range between 100k and 2M. If the upper limit is reached a warning is shown
+  and you may still need to press 'o' if you want to increase the refinement
+  even further.
```

Comment:
We're actually limiting the total number of vertices used for plotting the subdivided mesh. This is motivated by the observation that the vertices determine the size of the floating point data pushed to OpenGL, the other data is just connectivity information.

Also, it should be made clear that the choice of the limit number between 100k and 2M is chosen based on the order of the solution and the mesh and on the number of drawn elements.

Also, this limit is additionally restricted so that no more than 16 refinements are actually performed for auto-refinement.

Also, the order-based limit will be exceeded when the number of the drawn elements is big enough, i.e. when using refinement factor of 1 needs more vertices than the order-based limit.

Also, it is good to clarify that the "number of drawn elements" refers to the number of mesh elements for 2D meshes (including 2D meshes embedded in 3D) and to the number of boundary elements for 3D meshes.

Reply:
I improved the changelog a bit, so it should be more clear now 😉 I use the number of elements consistently with the comment above ☝️ . I am not sure if it addresses all points, but have a look.

Reply:
@tzanio , could you please reformulate it further? Also with the "vertices terminology" now 🙂 .